### PR TITLE
Refactor cards controllers

### PR DESCRIPTION
**Pull request by javierm**
## Objectives

* Make it easier to add cards in other sections
* Make card title mandatory so we don't have cards without titles
* Remove unneeded code

### CODE REVIEW

**Review by Senen — CHANGES_REQUESTED** [review]
Hi @javierm, nice refactoring and cleaning work 🧹 👏🏼 .

I found probably a missing spec, and also discovered a translation resolution error introduced because of the new model validation. I added a few commits to share a possible solution and some improvements to make the form look like other admin ones. Let me know what do you think.

The root of the error is the following line of the `admin/widget/cards/_form.html.erb` file:

```
<%= f.submit(t("admin.homepage.#{action_name}.#{@card.header? ? "submit_header" : "submit_card"}"), class: "button success") %>
```

Now when we submit a card without title the `action_name` part of the expression is resolved to [:create | :update] and we do not have translations for that entries so I decided to extract the form to a component so we can handle that kind of logic in the way we are doing these days.

File: app/models/abilities/administrator.rb [hunk 1]
```diff
@@ -88,6 +88,7 @@ def initialize(user)
       can :manage, SiteCustomization::Page
       can :manage, SiteCustomization::Image
       can :manage, SiteCustomization::ContentBlock
+      can :manage, Widget::Card
```
[the remark this answers is not in the document]
Comment (javierm) [reply app/models/abilities/administrator.rb:91]: Done! :+1:

File: app/views/admin/site_customization/cards/index.html.erb [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
   <div class="float-right">
     <%= link_to t("admin.site_customization.pages.cards.create_card"),
-        new_admin_widget_card_path(page_id: params[:page_id]), class: "button" %>
+        new_admin_site_customization_page_widget_card_path(@page), class: "button" %>
```
[the remark this answers is not in the document]
Comment (javierm) [reply app/views/admin/site_customization/cards/index.html.erb]: I don't know :sweat_smile:. We don't have clear criteria regarding multiline statements :thinking:. I've changed it so it looks as you suggested anyway.

**Comment by javierm** [comment]
> The root of the error is the following line of the `admin/widget/cards/_form.html.erb` file:
> 
> ```
> <%= f.submit(t("admin.homepage.#{action_name}.#{@card.header? ? "submit_header" : "submit_card"}"), class: "button success") %>
> ```

I've used the `admin_submit_action` helper as we do in other places. Does it look OK for you? I was going to use ActiveRecord model translations, but since we use a different text for header cards, it's a bit hard :thinking:.

**Review by Senen — APPROVED** [review]

**Pull request merged**